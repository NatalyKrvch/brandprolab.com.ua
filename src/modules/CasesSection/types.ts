import { CasesSectionType } from '@/sanity/lib/types';

export interface CasesSectionProps {
  casesData: CasesSectionType;
}
