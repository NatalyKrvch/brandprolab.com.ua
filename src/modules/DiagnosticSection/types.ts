import { DiagnosticSectionType } from '@/sanity/lib/types';

export interface DiagnosticSectionProps {
  diagnosticData: DiagnosticSectionType;
}
