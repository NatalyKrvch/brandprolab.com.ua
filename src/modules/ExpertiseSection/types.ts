import { ExpertiseSectionType } from '@/sanity/lib/types';

export interface ExpertiseSectionProps {
  expertiseData: ExpertiseSectionType;
}
