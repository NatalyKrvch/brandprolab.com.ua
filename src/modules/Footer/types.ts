import { FooterSectionType } from '@/sanity/lib/types';

export interface FooterProps {
  footerData: FooterSectionType;
}
