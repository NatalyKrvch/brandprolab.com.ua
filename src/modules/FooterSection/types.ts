export interface FooterSectionProps {
  offerURL: string;
}
