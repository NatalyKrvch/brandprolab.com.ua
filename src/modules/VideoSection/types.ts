import { VideoSectionType } from '@/sanity/lib/types';

export interface VideoSectionProps {
  videoData: VideoSectionType;
}
