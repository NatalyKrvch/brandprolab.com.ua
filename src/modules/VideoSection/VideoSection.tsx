import {
  Button,
  ExternalLink,
  Label,
  SprayBackground,
  Video,
} from '@/components';
import { normalizeImageURL, normalizeText } from '@/utils';

import type { VideoSectionProps } from './types';

const VideoSection = ({ videoData }: VideoSectionProps) => {
  const {
    title,
    label,
    description,
    buttonText,
    buttonLink,
    VideoLink,
    videoCover,
  } = videoData;

  return (
    <section className="flex flex-col items-center desktop:grid desktop:grid-cols-2 desktop:gap-24 desktop:px-36">
      <div className="flex flex-col items-center text-center desktop:items-start desktop:text-left">
        <Label className="mb-6 px-5 py-2 text-sm font-normal leading-none tracking-normal tablet:mb-10 tablet:text-lg desktop:mb-8 desktop:text-lg desktop:leading-none">
          {normalizeText(label)}
        </Label>

        <h2 className="mb-4 px-10 text-center text-32 font-bold leading-34 tracking-tight text-black tablet:mb-8 tablet:max-w-610 tablet:text-42 tablet:leading-none desktop:mb-8 desktop:max-w-456 desktop:px-0 desktop:text-left desktop:text-52 desktop:leading-54">
          {normalizeText(title)}
        </h2>

        <div className="tablet:max-w-480 relative mb-8 flex w-full max-w-400 justify-center desktop:hidden">
          <SprayBackground className="-left-30 -top-250 h-650 w-full -rotate-30 tablet:-left-30 tablet:-top-350 tablet:h-1000 tablet:w-full tablet:-rotate-45" />
          <Video
            videoUrl={VideoLink.trim()}
            thumbnailSrc={normalizeImageURL(videoCover)}
          />
        </div>

        <p className="mb-6 w-280 text-center text-sm font-normal leading-18 tracking-normal text-gray-dark tablet:w-480 tablet:max-w-482 tablet:text-lg tablet:leading-6 desktop:mb-14 desktop:w-456 desktop:text-left desktop:text-lg desktop:leading-6">
          {normalizeText(description)}
        </p>

        <ExternalLink
          aria-label="Перейти до запису на консультацію"
          className="w-200 tablet:w-314 desktop:w-400"
          href={buttonLink.trim()}
        >
          <Button>{normalizeText(buttonText)}</Button>
        </ExternalLink>
      </div>

      <div className="relative hidden w-full justify-center desktop:flex">
        <SprayBackground className="-rotate-30 desktop:-top-350 desktop:h-1020 desktop:w-full" />
        <Video
          videoUrl={VideoLink.trim()}
          thumbnailSrc={normalizeImageURL(videoCover)}
        />
      </div>
    </section>
  );
};

export default VideoSection;
