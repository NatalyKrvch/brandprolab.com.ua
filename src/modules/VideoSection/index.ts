export * from './types';
export { default as VideoSection } from './VideoSection';
