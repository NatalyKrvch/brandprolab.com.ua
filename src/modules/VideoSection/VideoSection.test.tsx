import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import { VideoSectionProps } from './types';
import VideoSection from './VideoSection';

jest.mock('@/sanity/lib/image', () => ({
  urlFor: () => ({
    url: () => '/test-thumbnail.jpg',
  }),
}));

jest.mock('@/sanity/lib/client', () => ({
  client: {
    fetch: jest.fn(),
  },
}));

describe('VideoSection component', () => {
  const mockData: VideoSectionProps['videoData'] = {
    _id: 'test-id',
    _type: 'videoSection',
    title: 'Test Video Section Title',
    label: 'Test Label',
    description: 'Test Description',
    buttonText: 'Test Button',
    buttonLink: '/test-link',
    VideoLink: 'https://youtube.com/watch?v=testvideo123',
    videoCover: {
      _type: 'image',
      asset: {
        _ref: 'image-someassetid',
        _type: 'reference',
      },
    },
  };

  it('renders VideoSection correctly', () => {
    const { asFragment } = render(<VideoSection videoData={mockData} />);

    expect(screen.getByText('Test Video Section Title')).toBeInTheDocument();
    expect(screen.getByText('Test Label')).toBeInTheDocument();
    expect(screen.getByText('Test Description')).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Test Button' })).toHaveAttribute(
      'href',
      '/test-link',
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
