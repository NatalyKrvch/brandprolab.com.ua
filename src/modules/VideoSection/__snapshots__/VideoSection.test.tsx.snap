// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoSection component renders VideoSection correctly 1`] = `
<DocumentFragment>
  <section
    class="flex flex-col items-center desktop:grid desktop:grid-cols-2 desktop:gap-24 desktop:px-36"
  >
    <div
      class="flex flex-col items-center text-center desktop:items-start desktop:text-left"
    >
      <label
        class="flex max-w-max items-center justify-center rounded-40 bg-teal-lightOpacity text-blue-dark mb-6 px-5 py-2 text-sm font-normal leading-none tracking-normal tablet:mb-10 tablet:text-lg desktop:mb-8 desktop:text-lg desktop:leading-none"
        data-testid="label-component-test-id"
      >
        Test Label
      </label>
      <h2
        class="mb-4 px-10 text-center text-32 font-bold leading-34 tracking-tight text-black tablet:mb-8 tablet:max-w-610 tablet:text-42 tablet:leading-none desktop:mb-8 desktop:max-w-456 desktop:px-0 desktop:text-left desktop:text-52 desktop:leading-54"
      >
        Test Video Section Title
      </h2>
      <div
        class="tablet:max-w-480 relative mb-8 flex w-full max-w-400 justify-center desktop:hidden"
      >
        <img
          alt=""
          aria-hidden="true"
          class="pointer-events-none absolute inset-0 z-[-1] select-none overflow-hidden object-cover blur-2xl -left-30 -top-250 h-650 w-full -rotate-30 tablet:-left-30 tablet:-top-350 tablet:h-1000 tablet:w-full tablet:-rotate-45"
          data-testid="spray-bg-test-id"
          src="/images/backgrounds/spray.svg"
        />
        <div
          class="relative h-210 w-280 overflow-hidden tablet:h-300 tablet:w-400 desktop:h-360 desktop:w-480 rounded-24 tablet:rounded-32 desktop:rounded-40"
          data-testid="video-test-id"
        >
          <div
            class="absolute inset-0 transition-opacity duration-500 opacity-100"
          >
            <div
              class="absolute inset-0 bg-teal-fogOpacity opacity-50"
            />
            <img
              alt=""
              aria-hidden="true"
              class="h-full w-full object-cover"
              data-nimg="1"
              data-testid="video-thumbnail-test-id"
              decoding="async"
              height="210"
              loading="lazy"
              src="/_next/image?url=%2Ftest-thumbnail.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Ftest-thumbnail.jpg&w=384&q=75 1x, /_next/image?url=%2Ftest-thumbnail.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="280"
            />
            <div
              class="absolute inset-0 flex items-center justify-center"
            >
              <button
                aria-label="Play video"
                class="rounded-full shadow-lg transition hover:scale-105"
                data-testid="video-play-button-test-id"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  data-nimg="1"
                  decoding="async"
                  height="74"
                  loading="lazy"
                  src="/icons/nav/play.svg"
                  style="color: transparent;"
                  width="74"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <p
        class="mb-6 w-280 text-center text-sm font-normal leading-18 tracking-normal text-gray-dark tablet:w-480 tablet:max-w-482 tablet:text-lg tablet:leading-6 desktop:mb-14 desktop:w-456 desktop:text-left desktop:text-lg desktop:leading-6"
      >
        Test Description
      </p>
      <a
        class="w-200 tablet:w-314 desktop:w-400"
        data-testid="external-link-test-id"
        href="/test-link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="w-full rounded-full py-3 text-xl font-medium leading-26 tracking-wide outline-none transition-all duration-300 hover:outline-offset-0 tablet:py-5 tablet:text-22 desktop:py-6 desktop:text-2xl desktop:leading-6 bg-teal-dark text-white hover:outline-8 hover:outline-teal-darkTranslucent active:bg-black"
          data-testid="button-test-id"
        >
          Test Button
        </button>
      </a>
    </div>
    <div
      class="relative hidden w-full justify-center desktop:flex"
    >
      <img
        alt=""
        aria-hidden="true"
        class="pointer-events-none absolute inset-0 z-[-1] select-none overflow-hidden object-cover blur-2xl -rotate-30 desktop:-top-350 desktop:h-1020 desktop:w-full"
        data-testid="spray-bg-test-id"
        src="/images/backgrounds/spray.svg"
      />
      <div
        class="relative h-210 w-280 overflow-hidden tablet:h-300 tablet:w-400 desktop:h-360 desktop:w-480 rounded-24 tablet:rounded-32 desktop:rounded-40"
        data-testid="video-test-id"
      >
        <div
          class="absolute inset-0 transition-opacity duration-500 opacity-100"
        >
          <div
            class="absolute inset-0 bg-teal-fogOpacity opacity-50"
          />
          <img
            alt=""
            aria-hidden="true"
            class="h-full w-full object-cover"
            data-nimg="1"
            data-testid="video-thumbnail-test-id"
            decoding="async"
            height="210"
            loading="lazy"
            src="/_next/image?url=%2Ftest-thumbnail.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Ftest-thumbnail.jpg&w=384&q=75 1x, /_next/image?url=%2Ftest-thumbnail.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="280"
          />
          <div
            class="absolute inset-0 flex items-center justify-center"
          >
            <button
              aria-label="Play video"
              class="rounded-full shadow-lg transition hover:scale-105"
              data-testid="video-play-button-test-id"
            >
              <img
                alt=""
                aria-hidden="true"
                data-nimg="1"
                decoding="async"
                height="74"
                loading="lazy"
                src="/icons/nav/play.svg"
                style="color: transparent;"
                width="74"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
